src/OSPSuite.SimModel/src/ExplicitFormula.cpp

```diff
@@ -363,6 +363,13 @@ void ExplicitFormula::Finalize()
 	for(int i =0; i<_quantityRefs.size(); i++)
 	{
 		QuantityReference * quantityRef = _quantityRefs[i];
+		//If the quantity is a species, add this formula to the list of formulas that use the species.
+		//Used for updating scale factors.
+		if (quantityRef->IsSpecies())
+		{
+			SimModelNative::Species * species = quantityRef->GetSpecies();
+			species->AddFormulaReference(this);
```

Comment:
this is not needed! The underlying formulas which use a species (VariableFormula, SimpleProductFormula) are added anyway.

Reply:
According to my tests, following classes **must** call `UpdateScaleFactorOfReferencedVariable` in order to pass the changing scale factor test in R https://github.com/Open-Systems-Pharmacology/R-Toolbox/issues/52 :

-ExplicitFormula
-IfFormula
-SimpleProductFormula
-Quantity

`VariableFormula` and `SimpleProductFormula` are not necessary. Now, the question is - are these really the only classes that require `UpdateScaleFactorOfReferencedVariable`, or are the others just not covered by tests? I wouldn't risk removing `UpdateScaleFactorOfReferencedVariable` from other classes until you provide a justification why they don't need it, and `IfFormula` does.